src/uct/cuda/base/cuda_md.c

```diff
@@ -262,6 +262,11 @@ ucs_status_t uct_cuda_base_mem_query(uct_md_h tl_md, const void *address,
         mem_attr->alloc_length = addr_mem_info.alloc_length;
     }
 
+    if (mem_attr->field_mask & UCT_MD_MEM_ATTR_FIELD_DMABUF_FD) {
```

Comment:
@yosefe 
1. rocm/cuda are the only two mds that I see supporting this. Not sure if I answered your question
```
$ grep -r "\.mem_query" src/uct/
src/uct/cuda/cuda_copy/cuda_copy_md.c:    .mem_query              = uct_cuda_base_mem_query,
src/uct/rocm/gdr/rocm_gdr_md.c:    .mem_query           = ucs_empty_function_return_unsupported,
src/uct/rocm/ipc/rocm_ipc_md.c:        .mem_query              = ucs_empty_function_return_unsupported,
src/uct/rocm/copy/rocm_copy_md.c:    .mem_query              = uct_rocm_base_mem_query,
src/uct/rocm/copy/rocm_copy_md.c:    .mem_query              = uct_rocm_base_mem_query,
```
2. Added FD_INVALID
3. When we introduced other attributes like base_addr, alloc_length, we didn't add new flags if an MD supports returning these attributes for a given memory type. The assumption was that if an MD can detect a certain memory type, it would support querying all attributes allowed by `uct_md_mem_query` API but would return valid values where possible. Can we not continue with the same logic for the new `fd` attribute?

Reply:
@Akshay-Venkatesh since there is no capability flag now, all other memory domains can return an undefined value when asked for UCT_MD_MEM_ATTR_FIELD_DMABUF_FD.
Should we document that only memory domains that are capable of memory detection are capable of detecting DMABUF?
How would UCP know which memory domain to query when registering DMABUF memory?

Reply:
@yosefe Seems like adding capability flag for MDs is better so that UCP explicitly checks instead of relying on documentation. After checking that MD is capable (for the given memory), we further query the dmabuf_fd. This could still return invalid but at least we don't have to call query on all MDs. Shall I add capability as a bit map?
```diff

    md_attr->cap.alloc_mem_types  = UCS_BIT(UCS_MEMORY_TYPE_CUDA) |
                                    UCS_BIT(UCS_MEMORY_TYPE_CUDA_MANAGED);
    md_attr->cap.access_mem_types = UCS_BIT(UCS_MEMORY_TYPE_CUDA) |
                                    UCS_BIT(UCS_MEMORY_TYPE_CUDA_MANAGED);
+    md_attr->cap.dmabuf_mem_types = UCS_BIT(UCS_MEMORY_TYPE_CUDA);
    md_attr->cap.detect_mem_types = UCS_BIT(UCS_MEMORY_TYPE_CUDA) |
                                    UCS_BIT(UCS_MEMORY_TYPE_CUDA_MANAGED);
```

Reply:
yes, let's add it

Reply:
@yosefe good catch. @Akshay-Venkatesh I agree that just documentation is not a good solution. CAP flag is the way to go.